bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHAApplianceHandler.java

```diff
@@ -0,0 +1,430 @@
+            if (!config.isValid()) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                        "Invalid configuration for Plugwise Home Automation appliance handler.");
+                return;
+            }
+            try {
+                PlugwiseHABridgeHandler bridge = this.getPlugwiseHABridge();
+                if (bridge != null) {
+                    PlugwiseHAController controller = bridge.getController();
+                    if (controller != null) {
+                        this.appliance = getEntity(controller, true);
+                        if (this.appliance != null) {
+                            if (this.appliance.isBatteryOperated()) {
+                                addBatteryChannels();
+                            }
+                            setApplianceProperties();
+                            updateStatus(ONLINE);
+                        }
+                    }
+                }
+            } catch (PlugwiseHAException e) {
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        super.thingUpdated(thing);
+
+        if (this.appliance.isBatteryOperated()) {
+            addBatteryChannels();
+        }
+
+        ThingHandler thingHandler = thing.getHandler();
+
+        if (thingHandler != null) {
+            for (Channel channel : thing.getChannels()) {
+                if (this.isLinked(channel.getUID())) {
+                    Appliance appliance = this.appliance;
+                    if (appliance != null) {
+                        this.refreshChannel(appliance, channel.getUID());
+                    }
+                }
+            }
+        }
+
+        setApplianceProperties();
+    }
+
+    @Override
+    protected @Nullable Appliance getEntity(PlugwiseHAController controller, Boolean forceRefresh)
+            throws PlugwiseHAException {
+        PlugwiseHAThingConfig config = getPlugwiseThingConfig();
+        Appliance appliance = controller.getAppliance(config.getId(), forceRefresh);
+
+        return appliance;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void handleCommand(Appliance entity, ChannelUID channelUID, Command command) throws PlugwiseHAException {
+        String channelID = channelUID.getIdWithoutGroup();
+
+        PlugwiseHABridgeHandler bridge = this.getPlugwiseHABridge();
+        if (bridge == null) {
+            return;
+        }
+
+        PlugwiseHAController controller = bridge.getController();
+        if (controller == null) {
+            return;
+        }
+
+        switch (channelID) {
+            case APPLIANCE_LOCK_CHANNEL:
+                if (command instanceof OnOffType) {
+                    OnOffType state = (OnOffType) command;
+                    try {
+                        if (state == OnOffType.ON) {
```

Comment:
If you don't use specific methods of the `OnOffType` it's not needed to put it into a variable, and you can directly use the `command` variable:
```suggestion
                    try {
                        if (command == OnOffType.ON) {
```